lib/matplotlib/axes/_axes.py

```diff
@@ -8519,6 +8532,19 @@ def violin(self, vpstats, positions=None, vert=True, widths=0.5,
         datashape_message = ("List of violinplot statistics and `{0}` "
                              "values must have the same length")
 
+        if vert or vert is False:
+            _api.warn_deprecated(
+                "3.9",
+                message=(
+                    "Passing `vert: bool` is deprecated. Support for this "
+                    "will be removed in Matplotlib %(removal)s. To change "
+                    "the orientation of the violin, use the "
+                    "`orientation: {'vertical', 'horizontal'}` parameter. "
+                )
```

Comment:
This message is missing `%(since)s`, but maybe instead of changing the message, this should just pass `alternative` (and any other relevant arguments to specify what's deprecated).